#Task 3 – Movie Genre Classification

##Problem Statement
###Build a Machine Learning model that predicts the genre of a movie based on its description.

##Approach
###Dataset: Movie Genre Dataset

##Text preprocessing
###Clean movie descriptions
###Remove stopwords and unnecessary characters

##Feature extraction
###Convert text into numerical features using TF-IDF

##Model Used
###Naive Bayes / Logistic Regression

##Steps
###Import dataset
###Clean and preprocess movie descriptions
###Convert text into TF-IDF features
###Train machine learning model
###Evaluate model performance
###Predict genre of new movie descriptions

##Result
###The model successfully classifies movies into their respective genres based on the description

In [27]:
import pandas as pd

# Load the dataset
data = pd.read_csv("train_data.txt", sep=':::', engine='python',
                   names=['ID','Title','Genre','Description'])

data.head()

,ID,Title,Genre,Description
0,1,Oscar et la dame rose (2009),drama,Listening in to a conversation between his do...
1,2,Cupid (1997),thriller,A brother and sister with a past incestuous r...
2,3,"Young, Wild and Wonderful (1980)",adult,As the bus empties the students for their fie...
3,4,The Secret Sin (1915),drama,To help their unemployed father make ends mee...
4,5,The Unrecovered (2007),drama,The film's title refers not only to the un-re...


In [12]:
# Remove rows where Genre or Description is missing
data = data.dropna(subset=['Genre', 'Description'])

In [13]:
data = data[['Genre','Description']]

data.head()

,Genre,Description
0,drama,Listening in to a conversation between his do...
1,thriller,A brother and sister with a past incestuous r...
2,adult,As the bus empties the students for their fie...
3,drama,To help their unemployed father make ends mee...
4,drama,The film's title refers not only to the un-re...


In [15]:
# Replace missing descriptions with empty text
data['Description'] = data['Description'].fillna('')

In [16]:
from sklearn.model_selection import train_test_split

X = data['Description']
y = data['Genre']

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

In [17]:
from sklearn.feature_extraction.text import TfidfVectorizer

vectorizer = TfidfVectorizer(stop_words='english', max_features=5000)

X_train_vec = vectorizer.fit_transform(X_train)
X_test_vec = vectorizer.transform(X_test)

In [18]:
from sklearn.naive_bayes import MultinomialNB

# Create model
model = MultinomialNB()

# Train model
model.fit(X_train_vec, y_train)

MultinomialNB()

In [19]:
predictions = model.predict(X_test_vec)

In [20]:
from sklearn.metrics import accuracy_score

print("Accuracy:", accuracy_score(y_test, predictions))

Accuracy: 0.46153846153846156


In [24]:
#examples testing
movie_description = ["A group of friends enter an abandoned asylum where evil spirits begin to hunt them one by one"]

movie_vec = vectorizer.transform(movie_description)
print("Predicted Genre:", model.predict(movie_vec)[0])

Predicted Genre:  horror 


In [25]:
movie_description = ["A clumsy man gets into hilarious situations while trying to impress his boss"]

movie_vec = vectorizer.transform(movie_description)
print("Predicted Genre:", model.predict(movie_vec)[0])

Predicted Genre:  comedy 


Conclusion

In this project, we built a movie genre classification system using TF-IDF vectorization and a Naive Bayes classifier.

The model was trained on movie plot descriptions and achieved an accuracy of about 46%.

We also tested the model using custom movie descriptions to predict their genres.